pages/data-providers/data-provider.jsx

```diff
@@ -1,7 +1,133 @@
-import React from 'react'
+import React, { useState, useEffect } from 'react'
+import { useRouter } from 'next/router'
 
 import DataProviderPageTemplate from 'templates/data-provider'
+import useDebouncedEffect from 'hooks/use-debounced-effect'
 
-const DataProviderPage = () => <DataProviderPageTemplate />
+let timeout = null
+
+// TODO: Pages should be just a thin layer of NextJS.
+//       Find a better place for it.
+const useDiscoveryService = (defaultDataProviderUrl) => {
+  const [dataProviderUrl, setDataProviderUrl] = useState(defaultDataProviderUrl)
+  const [discoveredUrls, setDiscoveredUrls] = useState(null)
+  const [isLoading, setIsLoading] = useState(false)
+
+  const handleDataProviderUrlChange = (newUrl) => {
+    setDiscoveredUrls(null)
+    setDataProviderUrl(newUrl)
+  }
+
+  useDebouncedEffect(() => {
+    if (dataProviderUrl === '') return
+
+    setIsLoading(true)
+
+    // TODO: Call API Here
+    clearTimeout(timeout)
+    timeout = setTimeout(() => {
+      if (dataProviderUrl === 'success.url')
+        setDiscoveredUrls([{ url: dataProviderUrl, isNew: true }])
+      else if (dataProviderUrl === 'duplicate.url')
+        setDiscoveredUrls([{ url: dataProviderUrl }])
+      else if (dataProviderUrl === 'multiple-duplicates.url')
+        setDiscoveredUrls([{ url: dataProviderUrl }, { url: dataProviderUrl }])
+      else setDiscoveredUrls([])
+
+      setIsLoading(false)
+    }, 1000)
+  }, [dataProviderUrl])
+
+  useEffect(() => {
+    setDataProviderUrl(defaultDataProviderUrl)
+  }, [defaultDataProviderUrl])
+
+  return [
+    dataProviderUrl,
+    discoveredUrls,
+    isLoading,
+    handleDataProviderUrlChange,
+  ]
+}
+
+const getHelperMessage = (dataProviderUrl, discoveredUrls) => {
+  if (dataProviderUrl === '' || discoveredUrls === null) {
+    return {
+      message:
+        'It can be any URL: homepage address, any data resource address,' +
+        ' OAI-PMH endpoint or RIOXX endpoint',
+      variant: 'normal',
+    }
+  }
+
+  if (discoveredUrls.length === 0) {
+    return {
+      message:
+        'We could not find any OAI-PMH endpoint.' +
+        ' Please try it with another URL.',
+      variant: 'error',
+    }
+  }
+
+  if (discoveredUrls.length === 1 && discoveredUrls[0].isNew) {
+    return {
+      message: `We found OAI-PMH endpoint at ${discoveredUrls[0].url}`,
+      variant: 'success',
+    }
+  }
+
+  if (discoveredUrls.length === 1) {
+    return {
+      message: 'The repository is already added.',
+      variant: 'success',
+    }
+  }
+
+  return {
+    message:
+      `${dataProviderUrl} and ${discoveredUrls.length} more are already` +
+      ' in our system. If you host more than one repository on the same domain,' +
+      ' please specify exact OAI-PMH endpoint',
+    variant: 'error',
+  }
```

Comment:
I would rather treat the first as an error. I thought you will find similarity between these too but I did not say it clearly here, sorry.

This should be the same message that conditionally adds 'N more' if there is more than 1. It's possible that we have one data provider added but there are 2 ones hosted under the URL. I such case we should notify the user to point full OAI-PMH endpoint.